### ISSUE TITLE
LinAlgError in Python 3.7

### ISSUE THREAD

**Issue by tgsmith61591** [tgsmith61591/pmdarima#74]
#### Description

In working towards supporting Python 3.7, I'm encountering a cryptic `LinAlgError` on the Canova-Hansen test that I can't replicate outside of CI environments.

#### Steps/Code to Reproduce

I can't reproduce outside of specific CI environments. Here is exactly the code being run in [the circle job](https://circleci.com/gh/tgsmith61591/pmdarima/373):

```yaml
  cpython37:
    docker:
      - image: python:3.7
    working_directory: ~/pmdarima
    steps:
    - checkout
    - run: make develop
    - run: make test-unit
```

And here is code to run the same exact code in your own docker container (which doesn't reproduce the error 😡):

```bash
$ docker pull python:3.7
$ docker run -it python:3.7 /bin/bash
$ git clone -b py37 https://github.com/tgsmith61591/pmdarima && cd pmdarima
$ make develop
$ make test-unit
```

#### Expected Results

1. This error shouldn't occur
2. If this error does occur, it should happen in both environments, but it doesn't.

I'll dig into the Canova-Hansen test to see if anything funky is going on, but this strikes me more as a numpy/scipy versioning issue than anything else... I believe this is also related to #1, which, funny enough, we dismissed as being due to Numpy versioning as well.

**PRs are of course always welcome**. You may want to branch the WIP branch I currently have going on `py37`, if you so choose..

**Comment by tgsmith61591** [comment]
Even more interesting is that it also happens in the python 3.6 test on Circle. These are only failing on tests that are run from the `Makefile`. Anyone want to track this one down?...

**Comment by aaronreidsmith** [comment]
I get the error when I run it in a local docker. Not sure how helpful that is, but I am looking into it. From what I have read ([here](https://stackoverflow.com/a/23534294/10696164) and [here](https://stackoverflow.com/a/22523229/10696164)), this is caused by NaN, or passing in negative weights. I am by no means a Data Scientist, could `m=366` in the tests be introducing those scenarios?

**Comment by tgsmith61591** [comment]
It shouldn't be. And didn't in older versions. Besides, it occurs in a scikit fit, so scikit should fail for NaNs, so I'm not convinced that's it.

Interesting it fails locally for you, though... Can you do a `pip freeze` in your failing container and post it here?

**Comment by aaronreidsmith** [comment]
```
root@0c43e6168cf4:/pmdarima# pip freeze
atomicwrites==1.2.1
attrs==18.2.0
certifi==2018.11.29
chardet==3.0.4
codecov==2.0.15
coverage==4.5.2
cycler==0.10.0
Cython==0.29.2
flake8==3.6.0
idna==2.8
kiwisolver==1.0.1
matplotlib==3.0.2
mccabe==0.6.1
more-itertools==4.3.0
numpy==1.15.4
pandas==0.23.4
patsy==0.5.1
pluggy==0.8.0
-e git+https://github.com/tgsmith61591/pmdarima@b44db6728b7189f17851c55477419ed63d5bbe77#egg=pmdarima
py==1.7.0
pycodestyle==2.4.0
pyflakes==2.0.0
pyparsing==2.3.0
pytest==4.0.2
pytest-cov==2.6.0
python-dateutil==2.7.5
pytz==2018.7
requests==2.21.0
scikit-learn==0.20.2
scipy==1.2.0
six==1.12.0
statsmodels==0.9.0
urllib3==1.24.1
```

**Comment by aaronreidsmith** [comment]
Don't know if this helps, but it also fails for me using a local 3.6 container

**Comment by tgsmith61591** [comment]
I have the same exact `pip freeze` in my passing container 😭 What is the output (header) of `make test` for you? Here's what I have:

```bash
===================================================================== test session starts =====================================================================
platform linux -- Python 3.7.1, pytest-4.0.2, py-1.7.0, pluggy-0.8.0 -- /usr/local/bin/python
cachedir: .pytest_cache
rootdir: /pmdarima, inifile: setup.cfg
plugins: cov-2.6.0
collected 86 items
```

**Comment by tgsmith61591** [comment]
I broke the tests into smaller pieces and it looks like it's only on the `m=366` test:

https://circleci.com/gh/tgsmith61591/pmdarima/386

Still doesn't explain why it passes in local containers for me, but not on Circle.

**Comment by aaronreidsmith** [comment]
Same output 😭

Failing 3.7
```
================================================ test session starts =================================================
platform linux -- Python 3.7.1, pytest-4.0.2, py-1.7.0, pluggy-0.8.0 -- /usr/local/bin/python
cachedir: .pytest_cache
rootdir: /pmdarima, inifile: setup.cfg
plugins: cov-2.6.0
collected 86 items
```  
  
  
Failing 3.6
``` . 
================================================ test session starts =================================================
platform linux -- Python 3.6.7, pytest-4.0.2, py-1.7.0, pluggy-0.8.0 -- /usr/local/bin/python
cachedir: .pytest_cache
rootdir: /pmdarima, inifile: setup.cfg
plugins: cov-2.6.0
collected 86 items
```

**Comment by tgsmith61591** [comment]
The only reason I can imagine is that it's related to a Numpy/scipy/cython version. This PR upgraded required Scipy to >=0.19 and Cython to >=0.29. Travis and Appveyor use more specific `requirements.txt`, and not the package-provided one. Circle is the only service using the `requirements.txt`, so it's probably one of these versions causing issues.

**Comment by tgsmith61591** [comment]
Circle has a fancy feature where you can re-run a failed workflow with SSH enabled, which allows you to actually remote into the box:

<img width="204" alt="screen shot 2018-12-20 at 5 51 35 pm" src="https://user-images.githubusercontent.com/6124706/50317193-e91f2180-047f-11e9-85f3-48a493f70670.png">

Running the two containers side-by-side, there were several things I wanted to compare. I'm getting the same GCC versions in both:

```bash
gcc --version
gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
```

I walked line-by-line through the code in both containers, and got to the same place, where it inexplicably failed:

```python
>>> lmch = LinearRegression().fit(R1, wts)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/site-packages/sklearn/linear_model/base.py", line 485, in fit
    linalg.lstsq(X, y)
  File "/usr/local/lib/python3.7/site-packages/scipy/linalg/basic.py", line 1241, in lstsq
    raise LinAlgError("SVD did not converge in Linear Least Squares")
numpy.linalg.linalg.LinAlgError: SVD did not converge in Linear Least Squares
```

The enraging bit is that `R1` and `wts` are 100% the same in both sessions! Now here's where things get really peculiar. After looking at [this question](https://stackoverflow.com/questions/23533144/numpy-linalg-linalg-linalgerror-svd-did-not-converge?rq=1), I wondered whether we needed to normalize the exogenous array prior to fitting:

```python
>>> from sklearn.preprocessing import StandardScaler
>>> lmch = LinearRegression().fit(StandardScaler().fit_transform(R1), wts)
>>> lmch
LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None,
         normalize=False)
```

... and inexplicably, **this works**. Which led me to test and verify that this works as well:

```python
>>> lmch = LinearRegression(normalize=True).fit(R1, wts)
>>> lmch
LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=True)
```

Even the model residuals appear to be the same... Setting the `CHTest` to always use `normalize=True` appears to allow the SVD to converge. But I don't really like that that's the proposed solution... call me paranoid, but it seems peculiar that we wouldn't have run into this until the dependency versions changed.

**I did confirm that the default value for `normalize` did not change in sklearn recently, so I'm out of ideas there**

**Comment by tgsmith61591** [comment]
Merged in #75
